flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionView.java

```diff
@@ -91,7 +91,11 @@ public AvailabilityWithBacklog getAvailabilityAndBacklog(int numCreditsAvailable
 
     @Override
     public Throwable getFailureCause() {
-        return parent.getFailureCause();
+        Throwable cause = parent.getFailureCause();
+        if (cause != null) {
+            return new ProducerFailedException(cause);
+        }
+        return null;
```

Comment:
Can you document this contract in the `ResultSubpartitionView#getFailureCause` java doc? That:

> subpartition view can decide whether the failure cause should be reported to consumer as failure (primary failure) or as cancellation (secondary failure). Secondary failure can be reported only if producer (upstream task) is guaranteed to failover. Beware: incorrectly reporting failure cause as primary failure, can hide the true cause of the failure from the user.

?

Reply:
OK, I will add it.